### PR TITLE
Shiqiu/support log streaming feature

### PR DESCRIPTION
**Pull request by ShichaoQiu**
support log streaming feature

### CODE REVIEW

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 1]
```diff
@@ -94,3 +94,39 @@ def validate_nodes_count(namespace):
     if namespace.instance_count is not None:
         if namespace.instance_count < 1 or namespace.instance_count > 20:
             raise CLIError('--instance-count must be in the range [1,20]')
+
+
+def validate_log_limit(namespace):
+    temp_limit = None
+    try:
+        temp_limit = int(namespace.limit)
```
[the remark this answers is not in the document]
Comment (ShichaoQiu) [reply src/spring-cloud/azext_spring_cloud/_validators.py]: Good idea, have done.

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 2]
```diff
@@ -94,3 +94,39 @@ def validate_nodes_count(namespace):
...
+
+def validate_log_limit(namespace):
+    temp_limit = None
+    try:
+        temp_limit = int(namespace.limit)
+    except:
+        raise CLIError('--limit must contains only digit')
+    if temp_limit < 0:
+        raise CLIError('--limit must be in the range [1,2048]')
+    namespace.limit = min(temp_limit, 2048) * 1024
```
[the remark this answers is not in the document]
Comment (ShichaoQiu) [reply src/spring-cloud/azext_spring_cloud/_validators.py]: Yeah, have refined the logic and will print an error message.

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 3]
```diff
@@ -975,3 +1031,49 @@ def get_logs_loop():
...
+    )
+    try:
+        if response.status != 200:
+            raise CLIError("Failed to connect to '{}' with status code '{}' and reason '{}'".format(
+                url, response.status, response.reason))
+        std_encoding = sys.stdout.encoding
+
+        for chunk in stream(response):
+            if chunk:
+                print(chunk.decode(encoding='utf-8', errors='replace')
```
[the remark this answers is not in the document]
Comment (ShichaoQiu) [reply src/spring-cloud/azext_spring_cloud/custom.py]: Thanks, we have changed it from `print` to `logger.warning`

**Comment by sophiaso** [comment]
Verified the functionalities.

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 4]
```diff
@@ -363,6 +368,57 @@ def app_get_build_log(cmd, client, resource_group, service, name, deployment=Non
...
+            raise CLIError(
+                "No production deployment found for app '{}'".format(name))
+        deployment = client.deployments.get(
+            resource_group, service, name, deployment_name)
+        if not deployment.properties.instances:
+            raise CLIError("No instances found for deployment '{0}' in app '{1}'".format(
+                deployment_name, name))
+        instances = deployment.properties.instances
+        if len(instances) > 1:
+            print("Mulitple app instances found:")
```
Comment (MyronFanQiu) [inline src/spring-cloud/azext_spring_cloud/custom.py]: Could we raise error here too?

Comment (ShichaoQiu) [reply src/spring-cloud/azext_spring_cloud/custom.py]: Thanks, have changed it.

**Pull request merged**